tests/rest/admin/test_user.py

```diff
@@ -514,6 +572,109 @@ def test_create_user(self):
+                raise HttpResponseException(
+                    channel.code, channel.result["reason"], channel.result["body"]
+                ).to_synapse_error()
+
+        # Sync to set admin user to active
+        _do_sync_for_user(self.admin_user_tok)
+
+        self.store.get_monthly_active_count = Mock(
+            return_value=defer.succeed(self.hs.config.max_mau_value + 1)
+        )
+
+        # Set MAU limit
+        self.get_failure(
+            handler.register_user(localpart="local_part"), ResourceLimitError
+        )
+
+        self.store.get_monthly_active_count = Mock(
+            return_value=defer.succeed(self.hs.config.max_mau_value + 1)
+        )
+        self.get_failure(
+            handler.register_user(localpart="local_part"), ResourceLimitError
+        )
+
+        # Register new user with admin API
+        url = "/_synapse/admin/v2/users/@bob:test"
+
+        # Create user
+        body = json.dumps({"password": "abc123", "admin": False})
+
+        request, channel = self.make_request(
+            "PUT",
+            url,
+            access_token=self.admin_user_tok,
+            content=body.encode(encoding="utf_8"),
+        )
+        self.render(request)
+
+        self.assertEqual(201, int(channel.result["code"]), msg=channel.result["body"])
+        self.assertEqual("@bob:test", channel.json_body["name"])
+        self.assertEqual(False, channel.json_body["admin"])
+
+    def test_create_user_limit_mau_passiv_admin(self):
+        """
+        Check that a new regular user is created successfully if MAU limit is reached.
+        Admin user was not active before creating user and creation fails.
+        """
+        self.hs.config.registration_shared_secret = None
+
+        handler = self.hs.get_registration_handler()
+
+        self.hs.config.limit_usage_by_mau = True
+        self.hs.config.max_mau_value = 2
+        self.hs.config.mau_trial_days = 0
+
+        # Set MAU limit
+        self.store.get_monthly_active_count = Mock(
+            return_value=defer.succeed(self.hs.config.max_mau_value + 1)
+        )
+        self.get_failure(
+            handler.register_user(localpart="local_part"), ResourceLimitError
+        )
+
+        self.store.get_monthly_active_count = Mock(
+            return_value=defer.succeed(self.hs.config.max_mau_value + 1)
+        )
+        self.get_failure(
+            handler.register_user(localpart="local_part"), ResourceLimitError
+        )
+
+        # Register new user with admin API
+        url = "/_synapse/admin/v2/users/@bob:test"
+
+        # Create user
+        body = json.dumps({"password": "abc123", "admin": False})
+
+        request, channel = self.make_request(
+            "PUT",
+            url,
+            access_token=self.admin_user_tok,
+            content=body.encode(encoding="utf_8"),
+        )
+        self.render(request)
+
+        self.assertEqual(500, int(channel.result["code"]), msg=channel.result["body"])
```

Comment:
help me out here: what is the difference between these tests and the ones further up at line 327 etc? what is the purpose of calling sync etc?

Perhaps some more comments to explain the test strategy and whwhat we're trying to do would be helpful.

Reply:
I have added more comments.
The difference is the `sync`.
If admin did a sync before mau limit is reached, the admin can send `POST` requests and create new users.
If admin did not a `sync` before limit is reached, the admin can not create new users, because the admin can not send `POST` requests. The admin is limited like a normal user.